core/src/main/java/io/fabric8/maven/core/access/ClusterAccess.java

```diff
@@ -72,6 +73,18 @@ public String getNamespace() {
         return namespace;
     }
 
+    /**
+     * Returns true if this cluster is a traditional OpenShift cluster with the <code>/oapi</code> REST API
+     * or supports the new <code>/apis/image.openshift.io</code> API Group
+     */
```

Comment:
@jstrachan Just a remark: The following logic doesn't quite fit the comment above which returns true only if the old openshift detection returns true **and** the API group is supported. 

This means also that by default for older OpenShift clusters a Kubernetes build mode is used which creates images against a Docker daemon (was s2i up to now)